File: Data Structures & Algorithms/minimum-interval-including-query/submission-2.py
class Solution:
    def minInterval(self, intervals: List[List[int]], queries: List[int]) -> List[int]:
        
        res = []
        for q in queries :
            minLength = -1
            for left,right in intervals :
                if (left <= q <= right) :
                    length = (right - left  + 1)
                    if minLength == -1 or length < minLength:
                        minLength = length
            res.append(minLength)
        return res

File: Data Structures & Algorithms/longest-common-prefix/submission-2.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {

        bool is_unmatched = false;
        int i = 0;
        while (i < strs[0].size()){
            for (const auto str : strs){
                if (strs[0][i] != str[i]){
                    string res (strs[0].begin(), strs[0].begin()+i);
                    return res;
                }
            }
            i++;
        }
        return strs[0];
       
        
    }
};